Azure/azure-cli
# {AKS} `az aks create/update` use a common truncate function for limiting rule group name to 260 characters

**Related command**

```az aks create -n kaveeshcli22 -g kaveeshcli --location westeurope --enable-azure-monitor-metrics --azure-monitor-workspace-resource-id "{full_id}" --grafana-resource-id "{full_id}"```

```az aks update -n kaveeshcli22 -g kaveeshcli --enable-azure-monitor-metrics --enable-windows-recording-rules```

```az aks create -n kaveeshcli22 -g kaveeshcli --disable-azure-monitor-metrics ```

**Description**

This PR is to address the following feedback comment -> https://github.com/Azure/azure-cli/pull/27488#discussion_r1384278929 to use a common truncate function for limiting the length of the recording rule group  name to 260 characters at most.

**Testing Guide**

The commands listed above should succeed in cluster create and update with the output containing something similar to (true or false depending on wether enabling or disabling) :

```
 "azureMonitorProfile": {
    "metrics": {
      "enabled": true,
      "kubeStateMetrics": {
        "metricAnnotationsAllowList": "",
        "metricLabelsAllowlist": ""
      }
    }
  }
```


---

This checklist is used to make sure that common guidelines for a pull request are followed.
